homeassistant/components/nest/config_flow.py

```diff
@@ -281,23 +366,39 @@ async def async_step_pubsub(
                     self._structure_config_title = generate_config_title(
                         device_manager.structures.values()
                     )
-
-                self._data.update(
-                    {
-                        CONF_SUBSCRIBER_ID: subscriber_id,
-                        CONF_CLOUD_PROJECT_ID: cloud_project_id,
-                    }
-                )
                 return await self._async_finish()
 
+        subscriptions = {}
+        try:
+            eligible_subscriptions = (
+                await self._admin_client.list_eligible_subscriptions(
+                    expected_topic_name=self._data[CONF_TOPIC_NAME],
+                )
+            )
+        except ApiException as err:
+            _LOGGER.error(
+                "Error talking to API to list eligible Pub/Sub subscriptions: %s", err
+            )
+            errors["base"] = "pubsub_api_error"
+        else:
+            subscriptions.update(
+                {name: name for name in eligible_subscriptions.subscription_names}
+            )
+        subscriptions[CREATE_NEW_SUBSCRIPTION_KEY] = "Create New"
         return self.async_show_form(
-            step_id="pubsub",
+            step_id="pubsub_subscription",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_CLOUD_PROJECT_ID, default=cloud_project_id): str,
+                    vol.Optional(
+                        CONF_SUBSCRIPTION_NAME,
+                        default=next(iter(subscriptions)),
+                    ): vol.In(subscriptions),
                 }
             ),
```

Comment:
I am not 100% sure on this, but you can pass a translation key to a selectselector. maybe we can use that to translate the create_new. But I am not sure if all keys that are in the options need to be in string.json, because ideally it shows the raw strings if its not found.

Not a blocker, just curious

Reply:
Yeah, I was thinking the same thing that it would be ideal to translate create new, but then still have it work for th other options.

```
Please select a topic:
- ( ) Use A
- ( ) Use B
- ( ) Use C
- ( ) Create a new topic
```

My assumption was that given some of the work to enforce translation keys existing, that we wouldn't want to do this. I will follow up and get a definitive answer though.